waspc/data/Generator/templates/server/src/config.js

```diff
@@ -13,27 +13,73 @@ const config = {
     env,
     port: parseInt(process.env.PORT) || 3001,
     databaseUrl: process.env.DATABASE_URL,
+    // This option is sometimes needed when running behind proxies/load balancers.
+    // For example, this is required for secure cookies to work on Heroku.
+    // Ref: https://expressjs.com/en/guide/behind-proxies.html
+    // For now, we only handle the boolean case.
+    trustProxies: undefined,
     {=# isAuthEnabled =}
-    auth: {
-      jwtSecret: undefined
-    }
+    session: {
+      cookie: {
+        name: process.env.SESSION_COOKIE_NAME || 'wasp_session',
+        secret: undefined,
+        maxAgeMs: parseInt(process.env.SESSION_COOKIE_MAX_AGE) || 7 * 24 * 60 * 60 * 1000, // 1 week in ms
+      },
+    },
+    csrf: {
+      cookie: {
+        name: process.env.CSRF_COOKIE_NAME || 'wasp_csrf',
+      },
+    },
     {=/ isAuthEnabled =}
+    frontendUrl: undefined,
   },
   development: {
+    trustProxies: parseBooleanOrDefault(process.env.TRUST_PROXIES, false),
     {=# isAuthEnabled =}
-    auth: {
-      jwtSecret: 'DEVJWTSECRET'
-    }
+    session: {
+      cookie: {
+        secret: process.env.SESSION_COOKIE_SECRET || 'sessionSecret',
+      },
+    },
     {=/ isAuthEnabled =}
+    frontendUrl: process.env.REACT_APP_URL || 'http://localhost:3000',
   },
   production: {
+    trustProxies: parseBooleanOrDefault(process.env.TRUST_PROXIES, true),
     {=# isAuthEnabled =}
-    auth: {
-      jwtSecret: process.env.JWT_SECRET
-    }
+    session: {
+      cookie: {
+        secret: process.env.SESSION_COOKIE_SECRET,
+      },
+    },
     {=/ isAuthEnabled =}
+    frontendUrl: process.env.REACT_APP_URL,
   }
 }
 
 const resolvedConfig = _.merge(config.all, config[env])
 export default resolvedConfig
+
+function parseBooleanOrDefault(str, defaultValue) {
+  if (!str) {
+    return defaultValue
+  }
+
+  switch(str.toLowerCase()) {
+    case "t":
+    case "true":
+      return true
+    case "f":
+    case "false":
+      return false
+    default:
+      return defaultValue
+  }
+}
+
+export function checkCookieSecretLength(secret) {
+  if (!secret || secret.length < 32) {
+    throw new Error("SESSION_COOKIE_SECRET must be at least 32 characters long in production")
```

Comment:
A bit of nitpicking but you are assuming in this error message that this function is called in production although you in theory don't know right here where it is called.
So maybe I would just remove the `in production` part.

Reply:
I see you call this function below, in another file. You could have called it directly here I think, in line 53, where you set the production.cookie.secret. You would need to modify it a bit maybe for ease of usage, but that way you would keep this logic here. I am not sure what is better though.

Reply:
Good catch, I can remove the "in production" and also see if we can use in this file instead. 👍🏻